src/core/framegraphes/forwardrenderer.cpp

```diff
@@ -885,6 +935,15 @@ void ForwardRenderer::setRenderSurface(QObject *renderSurface)
     m_surfaceSelector->setSurface(renderSurface);
 }
 
+/*!
+ * Sets the render target that the scene will be rendered into. If set to NULL,
+ * the scene will be rendered into a surface set by setRenderSurface
+ */
+void ForwardRenderer::setRenderTarget(Qt3DRender::QRenderTarget *renderTarget)
+{
+    m_renderTargetSelector->setTarget(renderTarget);
```

Comment:
shouldn't we rebuild the FrameGraph if we had previously set a window ?

Reply:
m_renderTargetSelector::targetChanged is connected to reconfigureFrameGraph() and handleSurfaceChange()